packages/client/src/stores/reducers/auth/authSlice.ts

```diff
@@ -0,0 +1,106 @@
+import {createSlice} from '@reduxjs/toolkit';
+import {RootState} from 'client/src/stores/store';
+import {transformUser} from 'client/src/api/apiTransformers';
+import {isUserData} from 'client/src/api/request/utilits';
+import {authThunks} from './authThunks';
+import {TAuthState} from './typing';
+
+const initialState: TAuthState = {
+	isLoading: false,
+	error: '',
+	user: null,
+};
+
+// Slice - функция, которая автоматически генерирует Action Creators и типы Action
+export const authSlice = createSlice({
+	name: 'auth',
+	initialState,
+	reducers: {
+	},
+	extraReducers: (builder) => {
+		builder
+		// fetchUser
+			.addCase(authThunks.fetchUser.fulfilled, (state, action) => {
+				state.isLoading = false;
+				if (isUserData(action.payload)) {
+					state.user = transformUser(action.payload);
+				}
+			})
+			.addCase(authThunks.fetchUser.pending, (state) => {
+				state.isLoading = true;
+				state.error = '';
+			})
+			.addCase(authThunks.fetchUser.rejected, (state, action) => {
+				state.isLoading = false;
+				state.error = action.error.message as string;
```

Comment:
`message` судя по типу `SerializedError` может быть `undefined`, не безопасно просто катануть ее к строке. На рантайме может выстрелить. Хоть здесь это и не сильно критично, стоит об это помнить и лучшей практикой будет явно прописать дефолтное значение, например так:
```suggestion
				state.error = action.error.message ?? 'Возникла неизвестная ошибка';
```